File: 1557-minimum-number-of-vertices-to-reach-all-nodes/1557-minimum-number-of-vertices-to-reach-all-nodes.cpp
class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) {
        vector<int> in(n);
        for (auto& edge : edges) ++in[edge.back()];
        
        vector<int> output;
        output.reserve(n);
        for (int i = 0; i < n; ++i) if (!in[i]) output.emplace_back(i);
        return move(output);
    }
};

File: 1557-minimum-number-of-vertices-to-reach-all-nodes/1557-minimum-number-of-vertices-to-reach-all-nodes.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        in_nodes = [0] * n
        for edge in edges:
            in_nodes[edge[1]] += 1
        return [i for i in range(n) if in_nodes[i] == 0]
        
        